From c0004f07b901099dea4083b50f4e25fa09970a2d Mon Sep 17 00:00:00 2001
From: Emil Velikov <emil.l.velikov@gmail.com>
Date: Sat, 21 Oct 2023 14:48:51 +0100
Subject: [PATCH 07/13] dkms: always use read_conf_or_die

Currently dkms status and by extension dkms autoinstall use read_conf,
which prints an error yet results in the commands succeeding.

Browsing across Arch and Debian, this can only happen on user error. As
such lets flip it to the fatal "_or_die" variant, like all the other
instances in the codebase.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
---
 dkms.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dkms.in b/dkms.in
index 881d083..43fcab6 100644
--- a/dkms.in
+++ b/dkms.in
@@ -1731,7 +1731,7 @@ do_status_weak()
 # interested in, but that the DKMS internals do not usually care about.
 module_status_built_extra() (
     set_module_suffix "$3"
-    read_conf "$3" "$4" "$dkms_tree/$1/$2/source/dkms.conf" 2>/dev/null
+    read_conf_or_die "$3" "$4" "$dkms_tree/$1/$2/source/dkms.conf" 2>/dev/null
     [[ -d $dkms_tree/$1/original_module/$3/$4 ]] && echo -n " (original_module exists)"
     for ((count=0; count < ${#dest_module_name[@]}; count++)); do
         tree_mod=$(compressed_or_uncompressed "$dkms_tree/$1/$2/$3/$4/module" "${dest_module_name[$count]}")
-- 
2.20.1

